src/main/java/com/shapesecurity/salvation/Tokeniser.java

```diff
@@ -98,7 +100,8 @@ private String next() {
                 continue;
             while (this.hasNext()) {
                 if (!this.eatDirectiveValue()) {
-                    throw this.createError("expecting directive-value but found " + this.next());
+                    int cp = this.next().codePointAt(0);
+                    throw this.createError(String.format("expecting directive-value but found %s (U+%04x). Non-whitespace characters outside ASCII 0x21-0x7E must be percent-encoded", this.next(), cp));
```

Comment:
Capitalise the message. Might also want to just say non-ASCII and non-printable characters. That's simpler than specifying the code point range.